swaylock/main.c

```diff
@@ -133,13 +171,23 @@ static void handle_global(void *data, struct wl_registry *registry,
 	} else if (strcmp(interface, zwlr_input_inhibit_manager_v1_interface.name) == 0) {
 		state->input_inhibit_manager = wl_registry_bind(
 				registry, name, &zwlr_input_inhibit_manager_v1_interface, 1);
+	} else if (strcmp(interface, zxdg_output_manager_v1_interface.name) == 0) {
+		state->zxdg_output_manager = wl_registry_bind(
+				registry, name, &zxdg_output_manager_v1_interface, 2);
 	} else if (strcmp(interface, wl_output_interface.name) == 0) {
 		struct swaylock_surface *surface =
 			calloc(1, sizeof(struct swaylock_surface));
 		surface->state = state;
 		surface->output = wl_registry_bind(registry, name,
 				&wl_output_interface, 3);
-		wl_output_add_listener(surface->output, &output_listener, surface);
+		wl_output_add_listener(surface->output, &wl_output_listener, surface);
+
+		struct zxdg_output_v1 *xdg_output =
```

Comment:
This will break if the server advertises wl_outputs before advertising xdg_output_manager, or if the server doesn't support xdg_output at all. You should poll xdg_outputs after the registry roundtrip.